airflow/models.py

```diff
@@ -3092,6 +3092,20 @@ def create_dagrun(self,
         run.refresh_from_db()
         return run
 
+    @provide_session
+    def get_dagruns(self, start_date=None, end_date=None, session=None):
+        """
+        Return a list of dag runs from this dag with execution date between
+        start_date and end_date
+        """
+        DR = DagRun
+        dr = session.query(DR).filter(DR.dag_id == self.dag_id)
+        if start_date:
+            dr = dr.filter(DR.execution_date >= start_date)
+        if end_date:
+            dr = dr.filter(DR.execution_date <= end_date)
```

Comment:
decided to keep it as is for now since 'airflow clear' cli is probably used in prod.